Alembic CurvesReader : Fix loading of indexed UVs

johnhaddon opened the pull request:
The call to `variableData()` resulted in this error :

```
Primitive::variableData() can only be used for non-indexed variables. Use Primitive::expandedVariableData() or access Primitive::variables directly. Primitive variable name: 'uv'
```

andrewkaufman reviewed (approved):
Seems good, worth adding a test?

johnhaddon commented:
> Seems good, worth adding a test?

But but but, if you test, you find bugs :)

Turns out we also had a problem _writing_ curve UV indices, so I've fixed that as well and included a test in 903bff7. I also added some convenience to the CurvesPrimitive constructor in ef44761, because we've learned quite a lot more about `boost::python` since writing the original bindings.
